projects/FW_102/src/main.c

```diff
@@ -0,0 +1,54 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define A_COUNTER_MS 500
+#define B_COUNTER_MS 1000
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_inc_counter(SoftTimerId timer_id, void *context);
+
+int main(void) {
+  // Initialize soft timer, and soft timer dependency (interrupt is needed)
+  interrupt_init();
+  soft_timer_init();
+
+  Counters loop_counter = {
+    .counter_a = 0,
+    .counter_b = 0,
+  };
+
+  // Increment counter a every 0.5s, counter b ever 1s.
+  soft_timer_start_millis(500, prv_inc_counter, &loop_counter, 0);
```

Comment:
I think you forgot to use the `X_COUNTER_MS` macros here and in the other `soft_timer_start_millis` call

Reply:
My bad! Must have rushed through everything